temporal-sdk/src/main/java/io/temporal/client/UpdateWithStartWorkflowOperation.java

```diff
@@ -0,0 +1,415 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.client;
+
+import io.temporal.common.Experimental;
+import io.temporal.workflow.Functions;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * UpdateWithStartWorkflowOperation is an update workflow request that can be executed together with
+ * a start workflow request.
+ */
+@Experimental
+public final class UpdateWithStartWorkflowOperation<R> {
+
+  private final AtomicBoolean invoked = new AtomicBoolean(false);
+
+  /**
+   * Returns a new builder for an {@link UpdateWithStartWorkflowOperation} for a zero argument
+   * request.
+   *
+   * @param request method reference to a proxy created through {@link
+   *     WorkflowClient#newWorkflowStub(Class, WorkflowOptions)}.
+   */
+  public static <R> Builder<R> newBuilder(Functions.Func<R> request) {
+    return new Builder<>(
+        () -> {
+          request.apply();
+        });
+  }
+
+  /**
+   * Returns a new builder for an {@link UpdateWithStartWorkflowOperation} for a one argument
+   * request.
+   *
+   * @param request method reference to a proxy created through {@link
```

Comment:
This method reference has to be a a method annotated as an Update correct? Can we clarify that in the docs here? I don't think it is immediately obvious looking at just this class what methods it can take

Reply:
good point! https://github.com/temporalio/sdk-java/pull/2199/commits/a05748d90f12edb44ec8451cf4cb81ecc31abd5a